src/core/utils/route-builder.ts

```
@@ -0,0 +1,175 @@
+    const toStation = graph.stations[toId];
+
+    const distance = calculateDistance(
+      fromStation.coordinates,
+      toStation.coordinates
+    );
+
+    // Get edge attributes (duration) from the graph if they exist
+    let segmentDuration = calculateTransitTime(distance);
+    try {
+      const edgeAttributes = graph.graph.getEdgeAttributes(fromId, toId);
+      if (edgeAttributes && edgeAttributes.duration) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)